File: back_end/opiniones-rest/Models/IRepositorioOpinion.cs
using System.Collections.Generic;
using Opinion.Modelo;

namespace Repositorio 
{
    public interface IRepositorioOpinion
    {
        string Create(OpinionModelo opinion);
        
        OpinionModelo FindById(string id);
        
        List<OpinionModelo> FindAll();
        
        bool Update(OpinionModelo opinion);
        
        bool Delete(string id);
        
        bool AddValoracion(string id, Valoracion valoracion);
    }
}

File: back_end/opiniones/src/main/java/repositorio/IRepositorioOpinion.java
package repositorio;

import java.util.List;

import modelos.Opinion;
import modelos.Valoracion;

public interface IRepositorioOpinion {
	
	String create(Opinion opinion);
    
	Opinion findById(String id);
    
	List<Opinion> findAll();
    
	boolean update(Opinion opinion);
    
	boolean delete(String id);
	
	boolean addValoracion(String id, Valoracion valoracion);
}
